source/glkernel/include/glkernel/sample.hpp

```diff
@@ -207,6 +208,33 @@ size_t poisson_square(tkernel<glm::tvec2<T, P>> & kernel, const T min_dist, cons
     return k + 1;
 }
 
+template <typename T, glm::precision P>
+size_t n_rooks(tkernel<glm::tvec2<T, P>> & kernel)
+{
+	assert(kernel.depth() == 1);
```

Comment:
same should be done in #1 - perhaps we can extend the signature to allow for an arbitrary "layer" at depth?